client/state/oauth2-clients/test/reducer.js

```diff
@@ -1,66 +1,35 @@
-/** @format */
-
 /**
  * External dependencies
  */
-import { expect } from 'chai';
 import { pick } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import reducer, { initialClientsData } from '../reducer';
-import {
-	OAUTH2_CLIENT_DATA_REQUEST,
-	OAUTH2_CLIENT_DATA_REQUEST_SUCCESS,
-	SERIALIZE,
-	DESERIALIZE,
-} from 'state/action-types';
+import { OAUTH2_CLIENT_DATA_RECEIVE } from 'state/action-types';
 
 describe( 'reducer', () => {
 	// Uses default data but reduces the size of this data set for tests
 	const initialState = pick( initialClientsData, [ 930, 973 ] );
 
-	test( 'should throw an error when no parameter is provided', () => {
-		expect( () => reducer() ).to.throw( TypeError );
-	} );
-
-	test( 'should throw an error when no action is provided', () => {
-		expect( () => reducer( {} ) ).to.throw( TypeError );
-	} );
-
-	test( 'should return the current state for an empty action', () => {
```

Comment:
Any particular reason for removing this and other tests? They don't add a whole bunch, but at least they test that the reducer doesn't behave unexpectedly in edge cases.

Reply:
The tests that test if the reducer throws error if called with no or invalid parameters: I think they don't do anything useful. Redux never calls them that way, so it doesn't matter at all whether they throw or not.

The "return current state for an empty action" test is testing the same trivial thing as the "return current state for an unknown action type" -- namely the `default: return state` line. That's what every reducer that's not completely broken will do.

Handling `SERIALIZE` and `DESERIALIZE` actions is no longer a job of this reducer. It was when it was implemented with `createReducer`, but now the persistence wrapper will be added by `combineReducer` by default. See [here](https://github.com/Automattic/wp-calypso/blob/8cf852867340c735948e9dfec78ebce34cf4cab7/client/state/utils/reducer-utils.js#L136) how the `setupReducerPersistence` wraps every plain reducer with `withoutPersistence` before combining.

Therefore, `SERIALIZE` and `DESERIALIZE` is fully covered by the `state/utils` tests.